### PR TITLE
EVG-2373 add js tests to self-tests

### PR DESCRIPTION
**Pull request by john-m-liu**
Patch: https://evergreen.mongodb.com/version/5a3931082fbabe2a16003e4a
I'm planning on parsing tests from this (EVG-2430) and fixing the tests (EVG-2431) in the next sprint

### CODE REVIEW

File: self-tests.yml [hunk 1]
```diff
@@ -800,3 +827,11 @@ buildvariants:
         distros:
           *generic-linux-distros
       - name: ".xc"
+
+  - name: js-tests
```
Comment (richardsamuels) [inline self-tests.yml]: Since you said they're broken, should we avoid running the tests for now?

Comment (bsamek) [reply self-tests.yml]: I agree. Maybe we should disable all test except the ones that pass? Or except a single one that noops?

**Comment by john-m-liu** [comment]
Removed the npm stuff from the self-tests

File: self-tests.yml [hunk 2]
```diff
@@ -432,6 +437,16 @@ tasks:
...
+    commands:
+      - command: git.get_project
+        type: system
+        params:
+          directory: gopath/src/github.com/evergreen-ci/evergreen
+      - func: run-make
+        vars:
+          target: "js-test"
+          build_env: "PATH=/opt/node/bin:$PATH"
```
Comment (tychoish) [inline self-tests.yml]: I'd just add this to the build variant's {{build_env} and then just do it like a normal task.

Comment (john-m-liu) [reply self-tests.yml]: Changed

File: self-tests.yml [hunk 3]
```diff
@@ -432,6 +437,15 @@ tasks:
...
             ./clients/linux_amd64/evergreen service deploy test-endpoints --commit ${revision} --username admin --key abb623665fdbf368a1db980dde6ee0f0
+  - name: js-tests
+    commands:
+      - command: git.get_project
+        type: system
+        params:
+          directory: gopath/src/github.com/evergreen-ci/evergreen
+      - func: run-make
+        vars:
+          target: "js-test"
```
Comment (tychoish) [inline self-tests.yml]: I would repalce this either with a new yaml anchor for run-js-test *or* just use the run go suite, which is basically, the same.

Comment (john-m-liu) [reply self-tests.yml]: done

**Pull request merged**